src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,66 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] amounts = getDataFromFile(fromFileName);
+        String[] resultData = buildData(amounts[0], amounts[1], amounts[2]);
+        writeData(resultData, toFileName);
+    }
+
+    private int[] getDataFromFile(String fileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line = reader.readLine();
+
+            while (line != null) {
+                if (line.split(",")[0].equals("supply")) {
+                    supplyAmount += Integer.parseInt(line.split(",")[1]);
+                } else if (line.split(",")[0].equals("buy")) {
+                    buyAmount += Integer.parseInt(line.split(",")[1]);
+                }
+
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data.", e);
+        }
+
+        int[] arrayWithAmounts = new int[3];
+        arrayWithAmounts[0] = supplyAmount;
+        arrayWithAmounts[1] = buyAmount;
+        arrayWithAmounts[2] = supplyAmount - buyAmount;
+
+        return arrayWithAmounts;
+    }
+
+    private String[] buildData(int amount1, int amount2, int amount3) {
```

Comment:
It  would be easiear if you just return string from here and then you can write whole string at once with bufferedWriter

Reply:
agree, done